src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -188,35 +190,26 @@ public ElasticSearchDao createIndexerInstance()
     ElasticSearchDao esDao = new ElasticSearchDao(getUri(), username, Secret.toString(password));
 
     esDao.setMimeType(getMimeType());
-    try {
-        esDao.setCustomKeyStore(getCustomKeyStore());
-    } catch (KeyStoreException | CertificateException |
-             NoSuchAlgorithmException | KeyManagementException | IOException e) {
-          LOGGER.log(Level.WARNING, e.getMessage(), e);
-    }
-    return esDao;
-  }
-
-  private KeyStore getCustomKeyStore() {
-    KeyStore customKeyStore = null;
-
-    // Fetch custom alias+certificate as a keystore (if present)
     if (!StringUtils.isBlank(customServerCertificateId)) {
-      StandardCertificateCredentials certificateCredentials = getCredentials(customServerCertificateId);
-      if (certificateCredentials != null) {
-        // Fetch keystore containing custom certificate
-        customKeyStore = certificateCredentials.getKeyStore();
+      try {
+          StandardCertificateCredentials certificateCredentials = getCredentials(customServerCertificateId);
+          if (certificateCredentials != null) {
+            esDao.setCustomKeyStore(certificateCredentials.getKeyStore(),
+                Secret.toString(certificateCredentials.getPassword()));
+          }
+      } catch (KeyStoreException | CertificateException |
+               NoSuchAlgorithmException | IOException e) {
+            LOGGER.log(Level.WARNING, e.getMessage(), e);
       }
     }
-
-    return customKeyStore;
+    return esDao;
   }
 
   private StandardCertificateCredentials getCredentials(String credentials)
   {
     return (StandardCertificateCredentials) CredentialsMatchers.firstOrNull(
-        CredentialsProvider.lookupCredentials(StandardCredentials.class,
-            Jenkins.getInstance(), ACL.SYSTEM, Collections.emptyList()),
+        CredentialsProvider.lookupCredentials(StandardCertificateCredentials.class,
+            Jenkins.get(), ACL.SYSTEM, Collections.emptyList()),
```

Comment:
shouldn't we use getOrDie() here?

Reply:
get() is the right method. These is no getOrDie() in Jenkins, here we can safely assume that Jenkins is up and running so this will never throw an exception. And getInstance() would return null when Jenkins is not up and running

Reply:
I thought there was one, but there isn't. 
I've checked that get() will throw on missing, that's what I was aiming for